content/post/190801-un-pas-vers-l-evolution-du-modele-bibliographique-rero-ils.md

```diff
@@ -0,0 +1,109 @@
+
+##### Implémentation
+
+Le catalogue RERO comporte environ 6,1 millons de manifestations, ce qui a fait quelque peu suer Bibliostratus lors de l'étape 1, du point de vue mémoire. Nous avons donc découpé le fichier en paquets de 500'000 notices. Cela nous a ensuite permis de lancer l'étape 2, la plus chronophage, en parallèle pour plusieurs fichiers, sans quoi ce traitement aurait pris plusieurs mois 😱. Le tout a été réalisé sur un PC, en voici un petit aperçu dans l'image ci-dessous:
+
+![Bibliostratus à l'œuvre sur l'un de nos PC](/img/bibliostratus_screen.png "Capture d'écran de Bibliostratus à l'œuvre sur l'un de nos PC")
+
+### Éviter les erreurs par une méthodologie stricte
+
+Avec les données de bibliothèque, la plupart des alignements automatiques basés sur des comparaisons de similarité génèrent des erreurs, que l'on qualifie de faux positifs: l'établissement d'une équivalence erronée.
+
+Pour réduire au maximum ces faux positifs, l'équipe RERO a testé Bibliostratus sur un échantillon de 10'000 notices représentatives du catalogue, puis a sélectionné et évalué les résultats. Le fruit de cette analyse a été communiqué aux personnes développant le logiciel, qui ont ensuite adapté Bibliostratus.
+
+Lors de l'étape de sélection, les données inutiles sont écartées, telles que celles ayant plusieurs équivalences pour une notice ou aucune équivalence du tout. Les données retenues sont ensuite triées par type d'algorithme de génération. Effectivement, Bibliostratus utilise selon les notices bibliographiques plus de 700 algorithmes différents pour établir des correspondances (il y a de la doc sur [Github](https://github.com/Transition-bibliographique/bibliostratus/wiki/2_annexe.-Le-m%C3%A9canisme-d'alignement)). Seuls ceux ayant généré le plus d'alignements sont retenus et évalués. A titre d'exemple, l'algorithme ayant créé le plus d'équivalences avec la BnF compare les ISBN puis les titres entiers (46% des cas).
+
+Les faux positifs les plus courants sont des cas où deux manifestations différentes de la même expression sont mises ensemble. Dans certaines situations, on soupçonne que les deux manifestations soient de fait les mêmes, mais que les données aient été saisies de manière différente. Cela peut s'expliquer par une divergence des règles de catalogage, une divergence d'analyse de la part du catalogueur ou une erreur humaine.
+
+Une fois la qualité d'une méthode validée, les résultats peuvent être intégrés en production, en sachant que le risque d'erreur ne peut être totalement écarté, notamment pour les documents sans ISBN. Toutefois, un faux positif ne sera en général pas totalement faux du fait qu'il regroupe quasi toujours deux manifestations de la même expression.
+
+### Quelques chiffres de l'alignement du catalogue RERO
+
+Les chiffres ci-dessous concernent uniquement les monographies textuelles du catalogue RERO, alignées avec la BnF de préférence, et à défaut avec le Sudoc.
+
+<table>
```

Comment:
For the table, I realize that we need also a shortcode to use the bootstrap classes. Thanks to [this blog post](https://zwbetz.com/style-a-markdown-table-with-bootstrap-classes-in-hugo/), it's not so difficult. Create a file in the `layout/shortcodes` named `bootstrap-table.html` with the following content:

```
<div class="table-responsive mt-4 mb-4">
{{ $htmlTable := .Inner | markdownify }}
{{ $class := .Get 0 }}
{{ $old := "<table>" }}
{{ $new := printf "<table class=\"%s\">" $class }}
{{ $htmlTable := replace $htmlTable $old $new }}
{{ $htmlTable | safeHTML }}
</div>
```
And then, the table can be written in MarkDown, inside the shortcode:

```
{{< bootstrap-table "table table-striped table-bordered" >}}
| Résultat de l'alignement     | Fréquence | % des monographies RERO | % de tout RERO |
|:-----------------------------|:---------:|:-----------------------:|:--------------:|
| Aucune équivalence           | 2'954'451 |          65.2%          |      48.1%     |
| 1 équivalence BnF            | 1'213'557 |          26.8%          |      19.8%     |
| 1 équivalence Sudoc          |  243'569  |           5.4%          |      4.0%      |
| Plusieurs équivalences BnF   |  108'693  |           2.4%          |      1.8%      |
| Plusieurs équivalences Sudoc |   12'896  |           0.3%          |      0.2%      |
{{< /bootstrap-table >}}
```
This `"table table-striped table-bordered"` are bootstrap classes found in the official documentation: https://getbootstrap.com/docs/4.3/content/tables/

On the second line of the table, the `:` are positioned to set the alignment.

The `<br>` is no more useful, as I have added `mt-4 mb-4` in the shortcode in order to had space.

![capture-19-08-06-11-27-02](https://user-images.githubusercontent.com/1557213/62528491-74d99880-b82c-11e9-8a68-ae739c26fcca.png)